MonoGame.Framework/Platform/Threading.cs

```diff
@@ -148,6 +150,41 @@ internal static void BlockOnUIThread<TState>(Action<TState> action, TState state
 #endif
         }
 
+        /// <summary>
+        /// Queues the action to occur on the UI thread and returns immediately.
+        /// </summary>
+        /// <param name="action">The action to be run on the UI thread</param>
+        internal static void OnUIThread(Action action)
+        {
+            if (action == null)
+                throw new ArgumentNullException("action");
+
+            OnUIThread(_metaAction, action);
+        }
+
+        /// <summary>
+        /// Queues the action to occur on the UI thread and returns immediately.
+        /// </summary>
+        /// <param name="action">The action to be run on the UI thread</param>
+        /// <param name="state">The data to pass to <paramref name="action"/></param>.
+        internal static void OnUIThread<TState>(Action<TState> action, TState state = default)
```

Comment:
@mrhelmut @harry-cpp

Note i added this so that some systems can make callbacks from worker thread that resolve on the main thread.

Specifically the callback for song ending comes from a native thread and we want the callback to occur from the main/UI thread.